tutorials/basic-topics/sorting/bucket-sort.md

```diff
@@ -1,14 +1,316 @@
 ---
+
 title: 'Bucket Sort'
-description: ''
+description: 'Bucket sort is a sorting algorithm that organizes elements into buckets based on their values and sorts each bucket separately for efficient sorting.'
 hide_table_of_contents: true
 draft: true
 keywords:
-  - leetcode
-  - tutorial
-  - sorting
-  - algorithm
-  - bucket sort
+
+- leetcode
+- tutorial
+- sorting
+- algorithm
+- bucket sort
+
 ---
 
-<TutorialAuthors names="@TBC"/>
+<TutorialAuthors names="@Sanchita1304"/>
+<Contributors names="@wingkwong"/>
+
+## Overview
+
+Bucket sort is a non-comparative sorting algorithm that works by distributing elements into a finite number of buckets, each representing a specific range of values. After distributing the elements, each bucket is sorted individually, typically using another sorting algorithm or recursively applying bucket sort. Finally, the sorted elements are concatenated to produce a fully sorted array. This algorithm is particularly useful when the input data is uniformly distributed across a range of values, making it efficient for certain types of data sets.
+
+## Algorithm
+
+1. Initialization:
+    
+    Create an array of empty buckets (containers), typically represented as lists or arrays.
+    Determine the range of input values and divide this range into equal-sized subranges, each corresponding to a bucket.
+    
+2. Distribution:
+    
+    Iterate through the input array and place each element into the appropriate bucket based on its value. This is done by mapping the       element's value to the corresponding bucket.
+    
+3. Sort Each Bucket:
+    
+    For each non-empty bucket, sort its contents using a sorting algorithm of your choice (e.g., insertion sort, quicksort, or another       bucket sort recursively).
+    
+4. Concatenation:
+    
+    Concatenate all the sorted buckets in order to obtain the final sorted array.
```

Comment:
the format could have been better. also don't know why there is some large spaces in 2 & 3.

Reply:
I have made changes to it and delete the spacing